api_spec.yaml

```diff
@@ -1266,17 +1490,370 @@ paths:
+                  required: true
+                  schema:
+                      type: string
+                      example: "google"
+                - name: authorization
+                  in: header
+                  required: true
+                  schema:
+                      type: string
+                      example: "Bearer API_KEY"
+            responses:
+                200:
+                    description: Success
+                    content:
+                        application/json:
+                            schema:
+                                type: object
+                                properties:
+                                    status:
+                                        type: string
+                                        default: "OK"
+                401:
+                    description: Unauthorised access
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+                                enum:
+                                    - Unauthorised access
+    /dashboard/api/tenants/third-party/boxy-saml/clients-with-saml-metadata:
+        put:
+            tags:
+                - Multitenancy
+            summary: Create or update client of Boxy SAML provider using SAML XML.
+            parameters:
+                - name: authorization
+                  in: header
+                  required: true
+                  schema:
+                      type: string
+                      example: "Bearer API_KEY"
+            requestBody:
+                description: Learn more about the options and how it works [here](https://supertokens.com/docs/thirdpartyemailpassword/common-customizations/saml/with-boxyhq/integration-steps).
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                thirdPartyIdSuffix:
+                                    type: string
+                                    description: Optional suffix for the thirdPartyId
+                                boxyURL:
+                                    type: string
+                                    description: Base URL of the SAML Jackson server
+                                boxyAPIKey:
+                                    type: string
+                                    description: API key of the SAML Jackson server
+                                encodedRawMetadata:
+                                    type: string
+                                    description: Base64 encoded XML metadata file. Not required if metadataURL is provided.
+                                metadataURL:
+                                    type: string
+                                    description: URL of the XML metadata file. Not required if encodedRawMetadata is provided.
```

Comment:
you could use some swagger option like oneOf / allOf, to make these 2 properties mutually exclusive.

Reply:
Done